Added documention about the header authentication (#341)
actualbudget/docs: docs/advanced/http-header-auth.md

Reviewer: This page is not linked anywhere and can only be found via search.

Update this file to add it to the sidebar: https://github.com/actualbudget/docs/blob/master/docs-sidebar.js

Or link to it from another page.

Author: good catch

# Authenticating With HTTP Headers

:::note

> Client Version 25.x.0 and

Reviewer 2: We should set this to be 25.6.0 which would be the June release.

Author: duh... it is the simple things that get you :p

Server Version 24.6.0 or higher are required for this feature.
:::

> This feature will allow Actual to use a HTTP header to automatically authenticate and login without prompting for a password. This would be useful for individuals who run SSO services like [Authentik](https://goauthentik.io/), [Authelia](https://www.authelia.com/), and more.

Reviewer: ```suggestion
This feature will allow Actual to use an HTTP header to automatically authenticate and log in without prompting for a password. This would be useful for individuals who run SSO services like [Authentik](https://goauthentik.io/), [Authelia](https://www.authelia.com/), and more.
```

## Setup

> This feature needs to be enabled on the server, it is not configure to work out of the box. In the actual config, set the value `loginMethod` or env `ACTUAL_LOGIN_METHOD` to `"header"`. This will enable header authentication, but the normal password authentication will still work as a fallback. 

Reviewer: ```suggestion
This feature needs to be enabled on the server, it is not configured to work out of the box. In the Actual config, set the value `loginMethod` or env `ACTUAL_LOGIN_METHOD` to `"header"`. This will enable header authentication, but the normal password authentication will still work as a fallback. 
```

> This feature needs to be enabled on the server, it is not configure to work out of the box. In the actual config, set the value `loginMethod` or env `ACTUAL_LOGIN_METHOD` to `"header"`. This will enable header authentication, but the normal password authentication will still work as a fallback. 

Reviewer: Should this maybe link to the other page describing `loginMehod`?

:::warning

> Be careful! A misconfiguration on this next step could make your instance available to whole internet. 

Reviewer: ```suggestion
Be careful! A misconfiguration on this next step could make your instance available to the whole internet. 
```

:::

> The SSO provider then needs to be configured to pass a extra HTTP header to actual. The details on how to do this are unique to the SSO provider, but the header `x-actual-password` needs to be set to your actual password.

Reviewer: ```suggestion
The SSO provider then needs to be configured to pass an extra HTTP header to Actual. The details on how to do this are unique to the SSO provider, but the header `x-actual-password` needs to be set to your actual password.
```

:::note

> This feature is not a HTTP basic auth, but a different form of using a password. For HTTP basic auth or User accounts see [this issue](https://github.com/actualbudget/actual/issues/524)

Reviewer: ```suggestion
This feature is not an HTTP basic auth, but a different form of using a password. For HTTP basic auth or user accounts see [this issue](https://github.com/actualbudget/actual/issues/524)
```

:::

## Configuration Options

* `loginMethod` (env: `ACTUAL_LOGIN_METHOD`): can be "password" (default) or "header". Must be set to "header" for this authentication option to work. Will fallback to password on header authentication failures

> * `trustedProxies` (env: `ACTUAL_TRUSTED_PROXIES`): List of CIDRs to allowed to send the authentication header. The default value for this is to allow only known internal ip address ranges. If your SSO provider is external, you will have to set this option to the CIDR/IP(s) of your SSO provider. If your provider is in your local network, this default option should work for you. 

Reviewer: This repeats the other page.  I would recommend removing it and providing a link to that page where it makes sense.
